### DISCUSSION TITLE
How to clean titles before adding to an entry list/ checking an entry list for duplicates and rejecting a download?

### DISCUSSION THREAD

**Discussion by zombiehoffa** [Flexget/Flexget#3804] [Q&A]
I'm using flexget to watch an rss feed and download stuff that matches keywords I add downloaded items to an entry list and then I compare to the entry list and reject stuff I've already downloaded. One thing I have noticed is the entry list is filling with titles that are not relevant to matching does anyone have some examples of using regexp and manipulate to clean titles in this context? I would need to clean a bunch of different release groups, mp4/x265, 1080p/720p, etc.

This seems easier in  a movie/tv show situation but I am doing it in a keyword matching context.

**Comment by gazpachoking** [comment]
The built in [seen](https://flexget.com/Plugins/seen) plugin is designed to reject things that have already been accepted before, you are trying to block more than the exact downloads? Are you only matching your keywords a single time? In that case it might be better to keep your list of desired keywords in a list (maybe a [regexp_list](https://flexget.com/Plugins/List/regexp_list)?) and have list_match remove them from the list once they've been matched.

**Reply by zombiehoffa** [reply]
It's a keyword matching scheme so I have it scrape the rss then look for a whole bunch of keywords and download if a keyword is hit. Seen is enabled, it just also doesn't work (ok maybe it's blocking exact matches but it's not blocking close matches, which is basically what I need) since it's searching stuff that doesn't fit  the movie/tv content scheme (think searching rss for an author as an idea of what the scrape is doing) as part of this I did the above but my entry list has the file name plus a bunch of stuff that isn't relevant to matching for this purpose (release group, mp4/x265, 720p,1080p, epub/mobi etc) and therefore it doesn't match and reject a download because said download has a different characteristic above.

I think this means I need to clean the irrelevant crap off the title before I add it to the list and before I compare a potential download to the list to make ti work, I just don't have a good example as to how to do that. I am also open to being out to lunch and someone filling me in on how to make seen work better for my use case, or some other solution that accomplishes the same thing.

**Reply by gazpachoking** [reply]
How about the second part of my question, if instead of having a giant static list of keywords you want, then storing a list of things you got to reject, you remove from the list of things you want as they are grabbed?

**Reply by zombiehoffa** [reply]
That would work if the list was specific things. It's more like looking for an author where I will probably want to download one of everything that pops up by that author but I don't want more than one and I don't want to not download anything new that pops up from the author.

**Reply by gazpachoking** [reply]
Ahh, gotcha. Yeah, it's a tough problem to solve then. I don't have any great ideas off hand.

**Reply by zombiehoffa** [reply]
I think I basically  need someone to help me with regexp to strip off 3 different things  before an item is added to the list and before an item is compared to the list.  I was really hoping someone had already done it so they could just copy/paste an example. The three things are:
1. release groups
2. file format (i.e. mp4, x265, HEVC, epub, mobi)
3. file quality (i.e. 720p, 1080p,4k,HD, etc)
